Sources/QuaternionModule/Quaternion.swift

```diff
@@ -358,8 +393,18 @@ extension Quaternion: Hashable {
     // representation. The correct behavior for zero falls out for free from
     // the hash behavior of floating-point, but we need to use a
     // representative member for any non-finite values.
+    // For any values not zero and infinity we use a canonical form, so
+    // that q and -q hash to the same value. This allows people who are using
+    // quaternions as rotations to get the expected semantics out of collections
+    // (while unfortunately producing some collisions for people who are not,
+    // but not in too catastrophic of a fashion).
```

Comment:
This is probably overkill. I think I would just canonicalize to `r.sign == .plus` and hash.

Reply:
I am sorry, I misinterpreted you comment on the sketch PR.
I guess I'll change it to hash the "canonicalized transform" representation then.

I do see/agree now, that the collisions for people who are not using quaternions for rotations are not too catastrophic :)